testing/test_hookrelay.py

```diff
@@ -76,3 +76,32 @@ def hello(self, arg):
     pm.register(Plugin())
     res = pm.hook.hello(arg=3)
     assert res == 4
+
+
+def test_defaults(pm):
+    """Verify that default keyword arguments can be declared on both specs
+    and impls. The default value look up precedence is up as follows:
+        - caller provided value
+        - hookspec default
+        - hookimpl default
+    """
+    class Api:
+        @hookspec
+        def myhook(self, arg, kwarg="default"):
```

Comment:
@nicoddemus yes this was the consensus prior as well.

My main concern with this is we may end up with lots of `if arg is None:` checks throughout `hookimpl`s. I guess it serves as an explicit way to show that your `hookimpl` may be further ahead then the defining project's `hookspec`?